### PR TITLE
Refactor GoogleSpeechRecognizer to support >=3.6

### PR DESCRIPTION
**Pull request by will-rice**
This change enables backward compatibility for Python 3.7.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def run(self) -> None:
         "License :: OSI Approved :: Apache Software License",
         "Operating System :: OS Independent",
     ],
-    python_requires=">=3.8",
+    python_requires=">=3.7",
```
Comment (space-pope) [inline setup.py:126]: Does it still require new features introduced in 3.7?

**Review by space-pope — COMMENTED** [review]
LGTM if it works without that initial blocking `get`. Also had a question about compatibility.

File: setup.py (hunk 1 above)
Comment (will-rice) [reply setup.py:126]: I don’t think so, but that’s something I will test. I'll see how far back we can go. There was also a build issue which doesn’t make a lot of sense so investigating that as well.

**Review by space-pope — APPROVED** [review]
LGTM. Thanks for widening the net!

**Pull request merged**